iter/iter.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+import random
+
+
+def gen_fasta(path: str):
+    body = []
+    stroka = str()
+    with open(path, 'r') as file:
+        body.append(file.readline().strip())
+        for line in file:
+            if line.startswith('>'):
+                body.append(stroka)
+                body.append(line.strip())
+                stroka = str()
+            else:
+                stroka += line.strip()
+    body.append(stroka)
+    for value in body:
+        yield value
+
+
+class GeneratorRex:
+    """
+    input: path to .fasta file
+    replacement - replace one or more letters in the given sequence
+    delit - remove a letter or piece in the given sequence
+    inser - insert one or more letters into the given sequence
+    chance - apply replacement, delit or inser with some probability to the passed sequence
+    """
+
+    def __init__(self, path_to_file):
+        self.path_to_file = path_to_file
+        self.box = []
+        self.stroka = str()
+        with open(self.path_to_file, 'r') as file:
+            file.readline()
+            for line in file:
+                if line.startswith('>'):
+                    self.box.append(self.stroka)
+                    self.stroka = str()
+                else:
+                    self.stroka += line.strip()
+        self.box.append(self.stroka)
+        self._coursor = -1
+
+    @staticmethod
+    def replacement(line):
+        lengh = len(line)
+        line_list = list(line)
+        amount_of_replace = range(int(lengh * round(random.choice([i*0.1 for i in range(5)]), 2)))
+        indexes_for_replace = [random.randint(0, lengh - 1) for _ in amount_of_replace]
+        for i in indexes_for_replace:
+            line_list[i] = random.choice(list(set(line)))  # replace to letter from line alphabet
+        return ''.join(line_list)
+
+    @staticmethod
+    def delit(line):
+        lengh = len(line)
+        rand_chance = random.randint(0, 3)  # rand_chance == 3 probability 1/4
+        if rand_chance == 3:  # big del
+            boundary_list = sorted([random.randint(0, lengh) for _ in range(2)])
+            return line[:boundary_list[0]] + line[boundary_list[1]:]
+        else:  # short del
+            boundary_num = random.randint(0, lengh)
+            return line[:boundary_num] + line[boundary_num + 1:]
+
+    @staticmethod
+    def inser(line):
+        lengh = len(line)
+        rand_chance = random.randint(0, 3)  # rand_chance == 3 probability 1/4
+        boundary = random.randint(0, lengh)
+        if rand_chance == 3:
```

Comment:
Вот так немного более интуитивно
```suggestion
        if random.random() < 0.25:
```